apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -317,11 +317,9 @@ where
             let context = request.context.clone();
             qp.plan(request).await.map(|response| {
                 if let Some(usage_reporting) = {
-                    context
-                        .extensions()
-                        .lock()
-                        .get::<Arc<UsageReporting>>()
-                        .cloned()
+                    let lock = context.extensions().lock();
+                    let urp = lock.get::<Arc<UsageReporting>>();
+                    urp.cloned()
```

Comment:
This is a no-op change. It was `cloned()` anyway.

Reply:
No.

```
When a temporary is returned in a function call in a match scrutinee, its lifetime lasts until the end of the match block, which may be surprising.
```

Locks will be held through the whole if / else scope (same with a match)